src/game/notechart.js

```diff
@@ -8,10 +8,15 @@ import GameNote   from './data/game-note'
 // game will ever need.
 //
 export class Notechart {
-  constructor(bmsNotes, timing, playerNumber, playerOptions) {
-    this._timing  = timing
-    this._notes   = this._generatePlayableNotesFromBMS(bmsNotes)
-    this._autos   = this._generateAutoKeysoundEventsFromBMS(bmsNotes)
+  constructor(bms, playerNumber, playerOptions) {
+    let bmsNotes    = BMS.Notes.fromBMSChart(bms)
+    let timing      = BMS.Timing.fromBMSChart(bms)
+    let keysounds   = BMS.Keysounds.fromBMSChart(bms)
+    this._timing    = timing
+    this._notes     = this._generatePlayableNotesFromBMS(bmsNotes)
+    this._autos     = this._generateAutoKeysoundEventsFromBMS(bmsNotes)
+    this._samples   = this._generateKeysoundFiles(keysounds)
+    this.keysounds  = keysounds.all()
```

Comment:
Consider move this up to line 15 and store the keysounds instance.

Create a getter.